src/main/java/com/redhat/darcy/salesforce/DateInputField.java

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright 2014 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of darcy-salesforce.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.redhat.darcy.salesforce;
+
+import com.redhat.darcy.ui.AbstractViewElement;
+import com.redhat.darcy.ui.annotations.Require;
+import com.redhat.darcy.ui.api.Locator;
+import com.redhat.darcy.ui.api.elements.DateInput;
+import com.redhat.darcy.ui.api.elements.Element;
+import com.redhat.darcy.ui.api.elements.Link;
+import com.redhat.darcy.ui.api.elements.Requireable;
+import com.redhat.darcy.ui.api.elements.TextInput;
+
+import static com.redhat.darcy.salesforce.RequiredInput.requiredInput;
+import static com.redhat.darcy.ui.By.xpath;
+import static com.redhat.darcy.ui.Elements.link;
+import static com.redhat.darcy.ui.Elements.textInput;
+import static com.redhat.darcy.web.By.htmlTag;
+
+import java.time.LocalDate;
+
+/**
+ * An ViewElement for a value that corresponds to a Date field on 
+ * a Salesforce object.
+ */
+public class DateInputField extends AbstractViewElement implements TextInput, 
+    DateInput, Link, Requireable {
+
+    @Require
+    private Element parent = super.parent;
+    
+    @Require
+    private TextInput nestedTextInput = textInput(byInner( 
+            xpath("//span[contains(@class,'dateInput')]"),
+            htmlTag("input")));
+    
+    @Require
+    private Link nestedLink = link(byInner(
```

Comment:
Yeah it's a little wonky.  The link has today's date [4/13/2015], rather than the word 'Today'.  Clicking it just puts that value into the date input field.  Naming things is hard :(